### PR TITLE
Oscar/data enhancements

### PR DESCRIPTION
**Pull request by Stereopoly**
* Notice on events page when none exist

Simulator Screen Shot - iPhone 11 Pro - 2020-06-08 at 10 31 10

### CODE REVIEW

File: bm-persona/Common/FilterTableView/FilterTableView.swift [hunk 1]
```diff
@@ -87,6 +99,24 @@ class FilterTableView<T>: UIView {
...
+        
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.topAnchor.constraint(equalTo: self.layoutMarginsGuide.topAnchor).isActive = true
+        view.leftAnchor.constraint(equalTo: self.layoutMarginsGuide.leftAnchor).isActive = true
+        view.rightAnchor.constraint(equalTo: self.layoutMarginsGuide.rightAnchor).isActive = true
+        view.bottomAnchor.constraint(equalTo: self.layoutMarginsGuide.bottomAnchor).isActive = true
+        
+        missingView = view
+    }
+}
```
Comment (huangshawn1) [inline bm-persona/Common/FilterTableView/FilterTableView.swift:117]: this code is almost identical to what's happening for the calendar. maybe make this a function in MissingDataView class that takes in a UIView to be added to (calendarCard for CalendarVC and self for FilterTableView)

File: bm-persona/Common/MissingDataView.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        label.text = "No items found"
+        
+        label.translatesAutoresizingMaskIntoConstraints = false
+        
+        label.center = self.center
+        
+        self.addSubview(label)
+        
+        label.centerXAnchor.constraint(equalTo: self.centerXAnchor).isActive = true
+        label.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
```
Comment (huangshawn1) [inline bm-persona/Common/MissingDataView.swift:31]: any reason to set the center and also constrain?

File: bm-persona/Common/MissingDataView.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    private var missingLabel: UILabel!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        let label = UILabel()
+        label.textColor = Color.darkGrayText
+        label.textAlignment = .center
+        label.font = Font.medium(18)
+        label.text = "No items found"
```
Comment (huangshawn1) [inline bm-persona/Common/MissingDataView.swift:22]: we should probably have a file to store all constant string messages like this one (for consistency in the future). not too relevant for this pr

Comment (huangshawn1) [reply bm-persona/Common/MissingDataView.swift:22]: for example "Nothing found!" when search has no results or "Unknown" when there's no phone number

File: bm-persona/Common/FilterTableView/FilterTableView.swift (hunk 1 above)
Comment (Stereopoly) [reply bm-persona/Common/FilterTableView/FilterTableView.swift:117]: Ah I remember what was happening - adding this view didn't work for some reason when using it in the FilterTableView

**Pull request merged**